core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -830,16 +830,14 @@ private void locationChanged(Item item, String oldName, String newName) {
             if (item instanceof AbstractProject) {
```

Comment:
Well this is wrong. Should not be limited to `AbstractProject` unless I am missing something.

Reply:
Next line tries to read the new AbstractProject, so this would have to be changed as well. What else should it be?
I am not sure for which types this ItemListener will be called. Does it make sense for anything else?

Reply:
Called for all item types, but what matters is the types that can have `FingerprintAction`s on their builds.

Otherwise the generalizing changes below don't accomplish anything.

Reply:
Since this is existing code and I am not sure how to change it correctly, could this be done separately?

Reply:
Just needs to check for `Job` rather than `AbstractProject` AFAIUI.

Reply:
Is the new version ok now?